### DISCUSSION TITLE
Do animations run in UI or JS thread in React Native?

### DISCUSSION THREAD

**Discussion by imsam67** [pmndrs/react-spring#1279] [Support] [answered]
I'm evaluating animation libraries for my React Native projects and I was wondering if animations produced by `react-spring` run in the UI or JavaScript thread in React Native projects.

One library I've looked at is `react-native-reanimated` and I understand, unlike the animation built-into `react-native`, the `react-native-reanimated` animations run in the UI thread which seems like the way to go for smoother animations. Which thread do `react-spring` animations run in `react-native` projects?

**Comment by aleclarson** [comment, marked as answer]
In the JS thread currently, but I plan to integrate Reanimated for my next project sometime this year.
